### ISSUE TITLE
Feature request: jsdoc/require-jsdoc full support for typescript

### ISSUE THREAD

**Issue by alejandroclaro** [gajus/eslint-plugin-jsdoc#484]
When using jsdoc/require-jsdoc with this configuration:

``` javascript
'jsdoc/require-jsdoc': [ 
    'error', 
   {
      'require': {
        'ArrowFunctionExpression': false,
         'ClassDeclaration': true,
        'FunctionDeclaration': true,
        'FunctionExpression': false,
        'MethodDefinition': true
      }
  }
],
```

for some reason, it does not complain about missing jdoc documentation in methods defined inside typescript's interfaces. For example:

``` typescript
/**
 * Foo interface documentation.
 */
export interface Foo extends Bar {
  /**
   * baz method documentation.
   */
  baz(): void;

  meow(): void;
}
```

Here `meow` is not documented but eslint does not detect this.

It would be nice to be able to control this, or just validate the interfaces in the same way as classes.

**Comment by brettz9** [comment]
You can add `TSMethodSignature` to the `contexts` array option (along with whatever `require`'s you need):

```js
'jsdoc/require-jsdoc': ['error', {
  contexts: [
      'TSMethodSignature',
  ],
  'require': {
    // ...
  }
}]
```

I've added your example to the docs (along with a corresponding passing version).

Note that if you need to find out the context of other places within JavaScript where you wish to enforce jsdocs, you can use https://astexplorer.net/ set to the `@typescript-eslint/parser` and click around to see the `type`.

Closing as that should address, but feel free to comment further as needed.

**Comment by alejandroclaro** [comment]
Thank @brettz9 It works, but it would be nice that this is considered in the default jsdoc/require-jsdoc when typescript is used. Maybe a new rule in typescript plugin.

**Comment by brettz9** [comment]
I think we could consider a PR which added it for our `settings.jsdoc.mode === 'typescript'`, but with workarounds possible, it is not a priority on my own list.

**Comment by alejandroclaro** [comment]
Thank @brettz9 

If we can find some time for this, we will try to contribute with a MR for this.

**Comment by ilyub** [comment]
Tried to add 
```ts
contexts: ['TSMethodSignature']
```
as suggested above.

The problem is that it is not enough to add it to jsdoc/require-jsdoc. You need to add it to every jsdoc rule (e.g. jsdoc/require-param). However not all rules have "contexts" configuration.

The same applies to checkConstructors, checkGetters, checkSetters - you need to add them each jsdoc rule, but not all rules have these settings.

It would be good global settings for context, checkConstructors, checkGetters, checkSetters

**Comment by brettz9** [comment]
> The problem is that it is not enough to add it to jsdoc/require-jsdoc. You need to add it to every jsdoc rule (e.g. jsdoc/require-param). However not all rules have "contexts" configuration.
> 
> The same applies to checkConstructors, checkGetters, checkSetters - you need to add them each jsdoc rule, but not all rules have these settings.
> 
> It would be good global settings for context, checkConstructors, checkGetters, checkSetters

Open to a PR which adds the support, even incrementally. Note that if the rules get options, one can "set-and-forget" some config which adds the same context everywhere and then just extend that config across project.

(Ideally, it would also be a good opportunity to ensure that `contexts` supported the same properties everywhere relevant, such as `tags` (to indicate that a context must have or must not have a certain tag on it).)
